background.js

```diff
@@ -24,7 +24,19 @@ chrome.tabs.onUpdated.addListener(function (tabId, changeInfo) {
 
   chrome.storage.local.get(["blocked", "enabled"], function (local) {
     const { blocked, enabled } = local;
-    if (Array.isArray(blocked) && enabled && blocked.find(domain => hostname.includes(domain))) {
+    if (!Array.isArray(blocked) || !enabled) {
+      return;
+    }
+
+    const allowList = blocked
+      .filter((item) => item.startsWith("!"))
+      .map((hostname) => hostname.substring(1));
```

Comment:
Suggesting to call this argument `item`, as map function will get, e.g.:
`!youtube.com`
which is not a `hostname` yet, but will be once `!` is stripped.